### DISCUSSION TITLE
Can't push yet again

### DISCUSSION THREAD

**Discussion by leblancr** [community/community#66905] [New to GitHub]
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.

Same old thing we've seen many times, so tired of seeing it. I've been able to push for a while then my token expired so I regenerated it now can't push. How can I fix this?

**Comment by khuramgill** [comment]
The error message you're seeing indicates that GitHub has removed support for password authentication when pushing to repositories over HTTPS. Instead, they now recommend using a personal access token for authentication. To fix this issue and push to your repository, you need to update your Git configuration to use a personal access token.

Here are the steps to fix this issue:

1. **Generate a Personal Access Token**:
   - Go to your GitHub account settings.
   - Click on "Developer settings" on the left sidebar.
   - Click on "Personal access tokens."
   - Click the "Generate token" button to create a new token.
   - Select the necessary scopes (permissions) for the token, such as "repo" for full repository access.
   - Click the "Generate token" button at the bottom.

2. **Update Your Git Configuration**:
   - Open a terminal or command prompt.
   - Use the following command to update your Git configuration with the personal access token. Replace `<your-token>` with the actual token you generated in step 1.
     ```
     git config --global credential.helper store
     git pull
     ```
     Git will prompt you to enter your username (usually your GitHub username) and your personal access token as the password. Make sure to enter your token instead of your GitHub password.

3. **Push to the Repository**:
   - After you've updated your Git configuration, you should be able to push to the repository using the token for authentication.
   - Use `git push` as usual to push your changes to the remote repository.

This should resolve the authentication issue, and you should be able to push to your GitHub repository using the personal access token. Remember to keep your personal access token secure and do not share it publicly, as it provides access to your GitHub account and repositories.
marked my answer

**Comment by leblancr** [comment]
Maybe I wasn't clear. When I said I regenerated my expired token that means I already have a token. 

git config --global credential.helper store does nothing and goes back to prompt.

git pull then says:
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> main

There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> main

git branch
* main

my branch is called 'main'.

$ git branch --set-upstream-to=origin/main main
fatal: the requested upstream branch 'origin/main' does not exist
hint: 
hint: If you are planning on basing your work on an upstream
hint: branch that already exists at the remote, you may need to
hint: run "git fetch" to retrieve it.
hint: 
hint: If you are planning to push out a new local branch that
hint: will track its remote counterpart, you may want to use
hint: "git push -u" to set the upstream config as you push.
hint: Disable this message with "git config advice.setUpstreamFailure false"
